### PR TITLE
Fix #194

### LINKED ISSUE #194: Plugin page crashes if there are huge number of tags or categories
We should not try to load all the tags/categories.

Requested at https://bulkwp.com/support/topic/bulk-delete-posts-page-timeout/

### PR DESCRIPTION
**Pull request by rajanit2000**

### CODE REVIEW

File: assets/js/src/bulk-delete.js [hunk 1]
```diff
@@ -6,12 +6,46 @@
...
 
-/*global BulkWP, postboxes, pagenow*/
+/*global BulkWP, postboxes, pagenow, ajaxurl*/
 jQuery(document).ready(function () {
 	/**
 	 * Enable select2
 	 */
-	jQuery( '.select2' ).select2();
+	jQuery( '.select2' ).select2({
+		width: '300px'
```
Comment (sudar) [inline assets/js/src/bulk-delete.js]: See if this can be included from css file.

File: assets/js/src/bulk-delete.js [hunk 2]
```diff
@@ -6,12 +6,46 @@
...
+					});
+ 
+				}
+				return {
+					results: options
+				};
+			},
+			cache: true
+		},
+		minimumInputLength: 3, // the minimum of symbols to input before perform a search
```
Comment (sudar) [inline assets/js/src/bulk-delete.js]: Change this to 2 characters.

File: bulk-delete.php [hunk 3]
```diff
@@ -441,6 +450,32 @@ public function add_delete_posts_settings_panel() {
...
+	 */
+	public function load_taxonomy_term(){
+		$response = array();
+
+		$taxonomy = sanitize_text_field( $_GET['taxonomy'] );
+
+		$terms = get_terms( array(
+			'taxonomy'   => $taxonomy,
+			'hide_empty' => false,
+			'search'     => $_GET['q'],
```
Comment (sudar) [inline bulk-delete.php]: Sanitize this field also

File: assets/js/src/bulk-delete.js (hunk 1 above)
Comment (rajanit2000) [reply assets/js/src/bulk-delete.js]: @sudar I could not set width using CSS, If i remove this line select2 automatically applied inline CSS

**Comment by rajanit2000** [comment]
@sudar i have done this changes

File: include/posts/class-bulk-delete-posts.php [hunk 4]
```diff
@@ -334,7 +343,14 @@ public static function render_delete_posts_by_tag_box() {
 			return;
 		}
 
-		$tags = get_tags();
+		$max_select2_value_tags = apply_filters( 'max_select2_value_tags', Bulk_Delete::MAX_SELECT2_LIMIT );
```
Comment (sudar) [inline include/posts/class-bulk-delete-posts.php]: Rename it to `bd_select2_ajax_limit`.

Add PHPDoc block with `@since` tag

**Comment by rajanit2000** [comment]
@sudar i have done this changes

**Pull request merged**